addons/test_mail/tests/test_message_post.py

```diff
@@ -109,7 +111,19 @@ def test_notify_by_mail_add_signature(self):
         self.assertEqual(found_mail.body_html.count(signature), 0)
 
     @users('employee')
-    def test_notify_by_email_prepare_rendering_contextt(self):
+    def test_notify_mail_add_signature_no_author_user_or_no_user(self):
+        self.test_message.author_id = self.env['res.partner'].sudo().create({
+            'name': 'Steve',
+        }).id
+        template_values = self.test_record._notify_by_email_prepare_rendering_context(self.test_message, {})
+        self.assertNotEqual(escape(template_values['signature']), escape('<p>-- <br/>Steve</p>'))
+
+        self.test_message.author_id = None
+        template_values = self.test_record._notify_by_email_prepare_rendering_context(self.test_message, {})
+        self.assertEqual(template_values['signature'], '')
+
+    @users('employee')
+    def test_notify_prepare_template_context_company_value(self):
```

Comment:
Its name was "test_notify_by_email_prepare_rendering_context" (probably due to resolving conflict)